metrics/bleurt/bleurt.py

```diff
@@ -0,0 +1,102 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" BLEURT metric. """
+
+import logging
+import os
+
+import nlp
+from bleurt import score  # From: git+https://github.com/google-research/bleurt.git
+
+
+logger = logging.getLogger(__name__)
+
+_CITATION = """\
+@inproceedings{bleurt,
+  title={BLEURT: Learning Robust Metrics for Text Generation},
+  author={Thibault Sellam and Dipanjan Das and Ankur P. Parikh},
+  booktitle={ACL},
+  year={2020},
+  url={https://arxiv.org/abs/2004.04696}
+}
+"""
+
+_DESCRIPTION = """\
+BLEURT a learnt evaluation metric for Natural Language Generation. It is built using multiple phases of transfer learning starting from a pretrained BERT model (Devlin et al. 2018)
+and then employing another pre-training phrase using synthetic data. Finally it is trained on WMT human annotations. You may run BLEURT out-of-the-box or fine-tune
+it for your specific application (the latter is expected to perform better).
+
+See the [README.md] file at https://github.com/google-research/bleurt for more information.
+"""
+
+_KWARGS_DESCRIPTION = """
+BLEURT score.
+
+Args:
+
+predictions` (list of str): prediction/candidate sentences
+`references` (list of str): reference sentences
+ checkpoint: BLEURT checkpoint. Will default to BLEURT-tiny if None.
+        
+Returns:
+    'scores': List of scores.
+"""
+
+CHECKPOINT_URLS = {
+    "bleurt-tiny-128": "https://storage.googleapis.com/bleurt-oss/bleurt-tiny-128.zip",
+    "bleurt-tiny-512": "https://storage.googleapis.com/bleurt-oss/bleurt-tiny-512.zip",
+    "bleurt-base-128": "https://storage.googleapis.com/bleurt-oss/bleurt-base-128.zip",
+    "bleurt-base-512": "https://storage.googleapis.com/bleurt-oss/bleurt-base-512.zip",
+    "bleurt-large-128": "https://storage.googleapis.com/bleurt-oss/bleurt-large-128.zip",
+    "bleurt-large-512": "https://storage.googleapis.com/bleurt-oss/bleurt-large-512.zip",
+}
+
+class BLEURT(nlp.Metric):        
+    def _info(self):
+
+        # check that config name specifies a valid BLEURT model
+        if self.config_name == 'default':
+            logger.warning("Using default BLEURT-Base checkpoint for sequence maximum length 128. "
+                           "You can use a bigger model for better results with e.g.: nlp.load_metric('bleurt', 'bleurt-large-512').")
+            self.config_name = "bleurt-base-128"
+        if self.config_name not in CHECKPOINT_URLS.keys():
+            raise KeyError(f"{self.config_name} model not found. You should supply the name of a model checkpoint for bleurt in {CHECKPOINT_URLS.keys()}")
+
+        # download the model checkpoint specified by self.config_name and set up the scorer
+        model_path = nlp.utils.file_utils.cached_path(CHECKPOINT_URLS[self.config_name], extract_compressed_file=True)
```

Comment:
@lhoestq is this the way we would like to have this or rather have a `DownloadManager` passed to `_info()`?
I feel like the second solution might cover more extensively future use cases and might be simpler to use.

Reply:
Yes we should use the `DownloadManager` as it will probably be used to:
- mock downloads when we're doing tests
- verify integrity of downloaded files
and it has/will have cool features (custom downloads, parallel downloads (not implemented yet) etc...)

Reply:
Maybe we could pass the `dl_manager` if `"dl_manager" in inspect.signature(self._info).parameters.keys()` for example, so that we don't have to change the signature of the other metrics (that would break backward compatibility).

Reply:
Or better: add another function called right after `_info()` that takes care of downloading the right metric resources: `_download_resources(self, dl_manager)`. By default this function would do nothing but could be implemented to download the model here

Reply:
I feel like the second solution (`_download_resources`) is the cleanest overall

Reply:
added a `download_and_prepare` function that is called during `load_metric`

similar to what happens to dataset, the parent class `Metric._download_and_prepare` does nothing and can be overridden by specific metrics

lmk how that looks

Note that there's no hash verification right now since it relies on `DatasetInfoDict`. we could create a similar `MetricInfoDict` class, but I figure we can do that in a future PR